doc/md/_a1_Getting-Started-GPIO.md

```diff
@@ -0,0 +1,222 @@
+
+```cpp
+void Init(Pin p, Mode m=Mode::INPUT, Pull pu=Pull::NOPULL, Speed sp=Speed::LOW);
+```
+
+The first thing you may be wondering, if you're not used to C++ is the `::` characters all over the place. These are used for the configuration settings so that the compiler will only allow you to use the correct inputs instead of being able to pass the wrong thing to the input, or even just mis-order the arguments, and end up scratching your head for hours wondering what's not working.
+
+These `::` are also used for namespaces, which we handle in the above example with the `using namespace` lines. Without that we'd have to use `daisy::GPIO` instead of `GPIO`, `daisy::seed::D0` instead of `D0`, etc.
+
+### Pin Argument
+
+So, the first argument: `Pin p` describes the daisy pin to assign to our GPIO, this ties our new GPIO object to some of the actual hardware on the Daisy Seed. By using `D0` (or more explicitly, `daisy::seed::D0`), we're saying we want to use that pin for our GPIO.
+
+### Mode Argument
+
+The second argument: `Mode m` can be any of the following:
+
+* `Mode::INPUT` - configures the pin as an input
+* `Mode::OUTPUT` - configures the pin as an output (in push pull configuration)
+* `Mode::OUTPUT_OD` - also an output, but the transistor connection to pull the signal to GND is not connected, this is less commonly used.
+* `Mode::ANALOG` - configures the GPIO for connection to the ADC or DAC peripherals within the Mircocontroller.
+
+If you only supply the `Pin` argument, the GPIO will default to `Mode::INPUT` configuration.
+
+### Pull Argument
+
+The third argument: `Pull pu` is used to select whether the GPIO will use an internal (around 30-50k) resistor as a pull up or pull down resistor.
+
+A pull up resistor will keep the GPIO idling at 3v3 unless something pulls it down, while a pull down resistor will keep a GPIO idling at 0V unless something pulls it up.
+
+The options for this argument are:
+
+* `Pull::NOPULL` - no pull up or pull down resistor is connected.
+* `Pull::PULLUP` - pull up resistor is connected to the GPIO line.
+* `Pull::PULLDOWN` - pull down resistor is connected to the GPIO line.
+
+This argument will default to `Pull::NOPULL` if you only supply a `Pin` and a `Mode`
+
+### Speed Argument
+
+The final argument, `Speed sp` is a special configuration for Output modes that control the slew rate, or the speed at which it can switch from 0V to 3V. We won't go super in depth on that in this, but you'll always want it to be the lowest speed possible for a given application.
+
+### Defaults
+
+So in our example above, we can avoid having to connect any external resistors to our button by using the pull up feature.
+
+If we wanted to use some external resistors we could pull one side high with a resistor to 5V, and the other with a resistor to GND, and connect one side to the GPIO input, and if we didn't need to use the GPIO pull up resistor we could initialize the pin using the default settings:
```

Comment:
This is a good point, and would probably benefit from a diagram instead of text.

Also, this was mentioned in a previous review, and I forgot to address it, but since the defaults are discussed on a per-parameter basis, this defaults section feels a bit redundant. 

So I'll probably just remove it.

Early next year we're going to go through and add diagrams for a lot of the hardware descriptions throughout these guides.